libusb/hid.c

```diff
@@ -1929,7 +1929,7 @@ uint16_t get_usb_code_for_current_locale(void)
 		return 0x0;
 
 	/* Make a copy of the current locale string. */
-	strncpy(search_string, locale, sizeof(search_string));
+	strncpy(search_string, locale, sizeof(search_string)-1);
```

Comment:
I don't believe this is required, is it?

Reply:
My local build compained. ;) Otherwise I woulnd't have change it.
```
/home/morty/data/cherry/hidapi/libusb/hid.c:1932:9: error: ‘__builtin_strncpy’ specified bound 64 equals destination size [-Werror=stringop-truncation]
 1932 |         strncpy(search_string, locale, sizeof(search_string));
      |         ^
cc1: all warnings being treated as errors
```

```
/usr/bin/cc --version
cc (Ubuntu 12.3.0-1ubuntu1~23.04) 12.3.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Reply:
I'd leave this, because this will probably pop up eventually when the tool chain is updated. Is that ok with you?

Reply:
ok